### PR TITLE
pkg/ansible: Updating reconciler to make errors more obvious

### LINKED ISSUE #1174: Ansible opeator playbook successfull but status still running with unknown reason code
When CR is created or reconcilled - the status or cr, and then play output are the following:

```
sh-4.2$ oc get cat operator-sample -o yaml | tail -n 18
status:
  conditions:
  - lastTransitionTime: 2019-03-06T10:08:25Z
    message: Running reconciliation
    reason: Running
    status: "False"
    type: Running
  - ansibleResult:
      changed: 13
      completion: 2019-03-06T10:08:57.041558
      failures: 0
      ok: 103
      skipped: 116
    lastTransitionTime: 2019-03-06T10:08:57Z
    message: non-zero return code
    reason: Failed
    status: "True"
    type: Failure
sh-4.2$ cat /tmp/ansible-operator/runner/ngp.cat.com/v1/catcustomresourcename/operator-sample/artifacts/latest/stdout | tail -n 5
META: ran handlers

PLAY RECAP *********************************************************************
localhost                  : ok=103  changed=13   unreachable=0    failed=0   

sh-4.2$ cat /tmp/ansible-operator/runner/ngp.cat.com/v1/catcustomresourcename/operator-sample/artifacts/latest/status
successfulsh-4.2$ cat /tmp/ansible-operator/runner/ngp.cat.com/v1/catcustomresourcename/operator-sample/artifacts/latest/rc
0
```

+
sometimes the message contains also:
```
      message: |-
        non-zero return code
        unknown playbook failure
```

The operator-sdk version:
```
cd ~/go/src/github.com/operator-framework/operator-sdk/ && git describe 
v0.5.0-50-gbf6eabe
```

Unfortunatelly there are no meaningful logs in the operator pod stdout.

PS. Nice feature by adding a symlink `latest` to the last ansible play logs!

### PR DESCRIPTION
**Pull request by shawn-hurley**
**Description of the change:**
* Taking a pass through the error scenarios and adding context to them
* Adding more obvious errors for end users around the reconcile period
errors
* Adding ability to handle ignore_error key in ansible task

**Motivation for the change:**
Better error messages and easier debugging when issues arise.

fixes #1174

### CODE REVIEW

File: pkg/ansible/controller/reconcile.go [hunk 1]
```diff
@@ -81,6 +82,9 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 	if ds, ok := u.GetAnnotations()[ReconcilePeriodAnnotation]; ok {
 		duration, err := time.ParseDuration(ds)
 		if err != nil {
+			// Should attempt to update to a failed condition
+			r.markError(u, request.NamespacedName, fmt.Sprintf("Unable to parse reconcile period: %v", err))
+			logger.Error(err, "Unable to parse reconcile period annotation")
```
Comment (jmrodri) [inline pkg/ansible/controller/reconcile.go]: Should the markError also mention `annotation`? If not, okay.

File: pkg/ansible/controller/reconcile.go [hunk 2]
```diff
@@ -96,6 +100,7 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 		u.SetFinalizers(finalizers)
 		err := r.Client.Update(context.TODO(), u)
 		if err != nil {
+			logger.Error(err, "Unable to upate cr with finalizer")
```
Comment (jmrodri) [inline pkg/ansible/controller/reconcile.go]: typo: upate -> update

File: pkg/ansible/controller/reconcile.go [hunk 3]
```diff
@@ -131,6 +137,8 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 
 	kc, err := kubeconfig.Create(ownerRef, "http://localhost:8888", u.GetNamespace())
 	if err != nil {
+		r.markError(u, request.NamespacedName, "Unable to run reconcilation")
+		logger.Error(err, "Unable to generate kubeconfig")
```
Comment (jmrodri) [inline pkg/ansible/controller/reconcile.go]: should the logger.Error mention `Unable to run reconciliation`? seems weird the messages are different.

TYPO: reconcilation -> reconciliation

File: pkg/ansible/controller/reconcile.go [hunk 4]
```diff
@@ -140,6 +148,8 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 	}()
 	result, err := r.Runner.Run(ident, u, kc.Name())
 	if err != nil {
+		r.markError(u, request.NamespacedName, "Unable to run reconcilation")
+		logger.Error(err, "Unable to run ansible runner")
```
Comment (jmrodri) [inline pkg/ansible/controller/reconcile.go]: Are the messages supposed to match? also TYPO: reconcilation -> reconciliation

File: pkg/ansible/controller/reconcile.go [hunk 5]
```diff
@@ -237,6 +248,47 @@ func (r *AnsibleOperatorReconciler) markRunning(u *unstructured.Unstructured, na
...
+	err := r.Client.Get(context.TODO(), namespacedName, u)
+	if apierrors.IsNotFound(err) {
+		logger.Info("Resource not found, assuming it was deleted", err)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	statusInterface := u.Object["status"]
+	statusMap, ok := statusInterface.(map[string]interface{})
```
Comment (jmrodri) [inline pkg/ansible/controller/reconcile.go]: Nice trick, didn't know about this.

File: pkg/ansible/controller/reconcile.go (hunk 3 above)
Comment (shawn-hurley) [reply pkg/ansible/controller/reconcile.go]: One is a user-facing error, one is for an admin to debug the operator. I don't think the user cares about the kubeconfig, the admin will.

**Pull request merged**